cylc/flow/terminal.py

```diff
@@ -199,6 +200,26 @@ def parse_dirty_json(stdout):
     raise ValueError(orig)
 
 
+def parse_reg(arg: str):
+    """Replace runN with true reg name.
+
+    Args:
+        arg (str): flow as entered by user on cli e.g. myflow/runN
+    """
+    arg = arg.rstrip('/')
+    if not arg.startswith(get_workflow_run_dir('')):
+        workflow_dir = get_workflow_run_dir(arg)
+    else:
+        workflow_dir = arg
+    run_number = re.search(
+        r'(?:run)(\d*$)',
+        os.readlink(workflow_dir)).group(1)
+    # importing here to counter circular dependancy errors.
+    from cylc.flow.workflow_files import WorkflowFiles
```

Comment:
Oh really, workflow_files imports from here? or is it more indirect?

Reply:
Yes, more indirect. Circle of dependencies can be seen pulling the import to the top... something like ```scripts/__init__ => terminal => workflow_files => remote => options_parser => terminal``` .
I'm just pushing a change to fix this, rather than leaving the import at time of use.